osxkeychain.go

```diff
@@ -146,29 +131,15 @@ func protocolTypeToGo(proto C.CFTypeRef) ProtocolType {
 	panic(fmt.Sprintf("unknown proto in protocolTypeToGo: %v", proto))
 }
 
-func authenticationTypeToC(t AuthenticationType) (at int) {
-	switch t {
-	case AuthenticationHTTPBasic:
-		at = C.kSecAuthenticationTypeHTTPBasic
-	case AuthenticationAny:
-		at = C.kSecAuthenticationTypeAny
-	default:
-		at = C.kSecAuthenticationTypeDefault
-	}
-	return
-}
-
-func authenticationTypeToGo(authtype C.CFTypeRef) AuthenticationType {
-	if authtype == nil {
-		// handle nil?
-		fmt.Println("nil authtype in authenticationTypeToGo")
-		return AuthenticationAny
-	}
-	switch authtype {
+func authenticationTypeFromRef(authType C.CFTypeRef) AuthenticationType {
+	// TODO: Fill this out.
+	switch authType {
 	case C.kSecAttrAuthenticationTypeHTTPBasic:
 		return AuthenticationHTTPBasic
+	case C.kSecAttrAuthenticationTypeDefault:
+		return AuthenticationDefault
 	}
-	panic(fmt.Sprintf("unknown authtype in authenticationTypeToGo: %v", authtype))
+	panic(fmt.Sprintf("unknown authType in authenticationTypeFromStringRef: %v", authType))
```

Comment:
I realize you didn't add this panic (or others in here) but we should probably remove them soon and try to handle this edge case more cleanly. Any thoughts on what the behavior of this should look like if the auth type is unknown?